### PR TITLE
feat: add additionalColumns prop to ContentExplorerModalContainer

### PR DESCRIPTION
**Pull request by apieceofbart**
What I'm trying to achieve here? We would like to use the current ContentExplorerModalContainer in admin console but we need to add extra column, namely access level column. The most generic way to do it without breaking changes is to allow developer to pass any extra column they wish - those would appear after the folder name column - I chose this arbitrary but I think it makes sense - you would most probably like to keep the folder name and icon first and the select checkbox last so any additional columns should probably go in between. We could and try to make the column order configurable but I think it wouldn't be very useful and it would complicate the code. 
Of course this is not the greatest solution but it's non breaking and it allows us to use ContentExplorerModalContainer as is which is a huge time saver.

Another idea we had was to use custom row renderer but I don't like it - it's hard to align the items in the extra column, the accessibility is breaking etc.

I am open to some other ideas as long as they achieve our goal.

image

### CODE REVIEW

**Review by tjuanitas — APPROVED** [review]
while I don't think it's ideal to pass multiple props that can serve similar purposes (in this case `itemRowRenderer`), I think this could be beneficial/useful for developers without having to fully customize their own rows.

i left some suggestions but they're non-blocking so good to me

File: src/features/content-explorer/item-list/ItemList.js [hunk 1]
```diff
@@ -287,6 +289,7 @@ ItemList.propTypes = {
     width: PropTypes.number.isRequired,
     height: PropTypes.number.isRequired,
     rowHeight: PropTypes.number,
+    additionalColumns: PropTypes.arrayOf(PropTypes.element),
```
[the remark this answers is not in the document]
Comment (apieceofbart) [reply src/features/content-explorer/item-list/ItemList.js:292]: Done

File: src/features/content-explorer/item-list/__tests__/ItemList.test.js [hunk 2]
```diff
@@ -267,4 +268,27 @@ describe('features/content-explorer/item-list/ItemList', () => {
...
+                    key="accessLevel"
+                    className="item-list-accessLevel-col"
+                    dataKey="accessLevel"
+                    flexGrow={1}
+                    flexShrink={0}
+                    width={0}
+                />,
+            ];
+            const wrapper = renderComponent({ items, additionalColumns });
+            expect(wrapper.find('.item-list-accessLevel-col')).toHaveLength(3);
```
[the remark this answers is not in the document]
Comment (apieceofbart) [reply src/features/content-explorer/item-list/__tests__/ItemList.test.js:291]: Sure

**Comment by apieceofbart** [comment]
@tjuanitas 
> while I don't think it's ideal to pass multiple props that can serve similar purposes (in this case itemRowRenderer), I think this could be beneficial/useful for developers without having to fully customize their own rows.

I agree but I don't think `itemRowRenderer` should be used to add new columns - there's a lot that the Table component is doing behind the scenes (adding markup, classnames, tab index etc.) - you could probably copy all that when using `itemRowRenderer` but that feels veeery hacky

**Review by tjuanitas — APPROVED** [review]

**Review by menaczar — APPROVED** [review]

**Review by SkyisAakash — APPROVED** [review]

**Pull request merged**